src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -5878,7 +5879,7 @@ protected override object PropertyGet(PSProperty property)
         protected override void PropertySet(PSProperty property, object setValue, bool convertIfPossible)
         {
             DataRowView dataRowView = (DataRowView)property.baseObject;
-            dataRowView[(string)property.adapterData] = setValue;
+            dataRowView[(string)property.adapterData] = PSObject.Base(setValue);
```

Comment:
Could we make PSObject more flexible/smart with IConvertable?
The error is raised in https://github.com/dotnet/runtime/blob/1c442fcb36e73eb3d8be10423a45389ce2a468b9/src/libraries/System.Data.Common/src/System/Data/Common/DateTimeStorage.cs#L165
The pattern looks as commonly used. So perhaps we could benefit from implementation IConvertable in PSObject.

Reply:
This sounds like a good idea, better than special-casing the `DataRow` and `DataRowView` in the binder code.